src/mesh/tdycoremesh.c

```diff
@@ -369,6 +369,9 @@ PetscErrorCode TDyAllocateMemoryForMesh(TDy tdy) {
   mesh->num_subcells = cNum*num_subcells;
   ierr = AllocateMemoryForSubcells(cNum, num_subcells, subcell_type, &mesh->subcells); CHKERRQ(ierr);
 
+  TDy_region *region = &mesh->region_connected;
```

Comment:
It might be nice, for this and for the other structs here (edges, faces, etc) to provide functions for creating and manipulating their state, and for destroying it when we're done. In the case of the `TDy_region` (which we could rename to `TDyRegion` to be more consistent with the other types and with the PETSc way of naming things:

```
// Creates a new empty region to which mesh cells can be added
PetscError  TDyRegionCreate(TDyRegion* region) {
  region->num_cells = 0;
  region->id = NULL;
  return 0;
}

// Adds the cells with the given IDs to the region.
PetscErrorCode TDyRegionAddCells(TDyRegion* region,
                                                         PetscInt numCells,
                                                         PetscInt cellIDs[numCells]) {
   region->num_cells += numCells;
   PetscError err = PetscRealloc(region->num_cells, &region->id);
   if (err != 0) return err;
   err = PetscMemcpy(region->id, cellIDs);
   return err;
}

// Destroys a region, deallocating all resources.
void TDyRegionDestroy(TDyRegion* region) {
  if (region->id != NULL) {
    TDyDeallocate_IntegerArray_1D(region->id);
  }
}
```

And so on and so forth. This isolates the code specific to a region and allows you to understand where you're doing allocations and all that.

Reply:
@jeff-cohere  Thanks for this example. In the coming days, I will refactor this PR for regions using the above example.